File: 프로그래머스/연습문제/Lv2/시소 짝꿍/시소 짝꿍.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

long long solution(vector<int> weights) {
    sort(weights.begin(), weights.end());
    
    long long cnt = 0;
    for(int i=0; i<weights.size()-1; i++){
        for(int j=i+1; j<weights.size(); j++){
            if(weights[i] == weights[j]) cnt++;
            else if(weights[j] == weights[i]*2) cnt++;
            else if(weights[i]*3 == weights[j]*2) cnt++;
            else if(weights[i]*4 == weights[j]*3) cnt++;
        }
    }
    
    return cnt;
}

File: 프로그래머스/연습문제/Lv2/시소 짝꿍/시소 짝꿍.py
from collections import Counter

def solution(weights):
    cnt = 0
    cnts = Counter(weights)
    
    for k, v in cnts.items():
        if v >= 2:
            cnt += v*(v-1)//2
    weights = set(weights)
    
    for w in weights:
        if w*2 in weights:
            cnt += cnts[w]*cnts[w*2]
        if w*3/2 in weights:
            cnt += cnts[w]*cnts[w*3/2]
        if w*4/3 in weights:
            cnt += cnts[w]*cnts[w*4/3]

    return cnt
